osu.Game/Beatmaps/BeatmapSetInfo.cs

```diff
@@ -59,6 +59,8 @@ public class BeatmapSetInfo : IHasPrimaryKey, IHasFiles<BeatmapSetFileInfo>, ISo
 
         public string StoryboardFile => Files?.Find(f => f.Filename.EndsWith(".osb", StringComparison.OrdinalIgnoreCase))?.Filename;
 
+        public string GetPathForFile(string filename) => Files?.SingleOrDefault(f => string.Equals(f.Filename, filename, StringComparison.OrdinalIgnoreCase))?.FileInfo.StoragePath;
```

Comment:
Best to add xmldoc for this (generally we try and xmldoc anything that is `public`) with a mention that the returned path is relative to the user file storage.